swift/NameNormalizerSwift/NameNormalizer/AuthorNameNormalizer.swift

```diff
@@ -1,3 +1,41 @@
+public enum AuthorNameNormalizerError: Swift.Error {
+    case multipleCommas
+}
+
 public func normalize(name: String) throws -> String {
-    fatalError("normalize not implemented")
+    let suffixParts = try splitSuffix(name)
+    let parts = suffixParts.nameWithoutSuffix.split(separator: " ")
+    if parts.count < 2 {
+        return suffixParts.nameWithoutSuffix
+    }
+    return "\(lastName(parts)), \(firstName(parts))\(middleInitial(parts))\(suffixParts.suffix)"
+}
+
+private func splitSuffix(_ name: String) throws -> (nameWithoutSuffix: String, suffix: String) {
+    let parts = name.split(separator: ",")
+    if parts.count < 2 {
+        return (nameWithoutSuffix: name, suffix: "")
+    } else if parts.count == 2 {
+        return (nameWithoutSuffix: String(parts.first!), suffix: ",\(parts.last!)")
+    } else {
+        throw AuthorNameNormalizerError.multipleCommas
+    }
+}
```

Comment:
I realize this is a stylistic preference... if I have an early return (or throw) in a small, well-composed method, I generally don't bother with the redundant "else" clause--mostly because I think it reads better. Up to you.

Reply:
Yeah, I usually do that. Good call.